backends/s3/src/main/scala/geotrellis/s3/S3Client.scala

```diff
@@ -14,10 +14,10 @@ import com.amazonaws.ClientConfiguration
 import org.apache.commons.io.IOUtils
 import scala.collection.JavaConversions._
 
-trait S3Client extends LazyLogging {
+trait S3Client extends LazyLogging with Serializable {
```

Comment:
I do not understand the point about moving the logic to a companion object.  Can you give me some idea of what that might look like?